catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java

```diff
@@ -0,0 +1,180 @@
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.codice.ddf.registry.schemabindings.helper.InternationalStringTypeHelper;
+import org.codice.ddf.registry.schemabindings.helper.MetacardMarshaller;
+import org.codice.ddf.registry.schemabindings.helper.SlotTypeHelper;
+import org.codice.ddf.security.common.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ExtrinsicObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.VersionInfoType;
+
+/**
+ * Creates a registry identity node when DDF Registry is first instantiated.
+ * If a previous registry node is not already found, then a registry identity metacard and its
+ * attributes are created. FederationAdminService adds the newly created identity metacard as a registry entry.
+ */
+public class IdentityNodeInitialization {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IdentityNodeInitialization.class);
+
+    private FederationAdminService federationAdminService;
+
+    private InputTransformer registryTransformer;
+
+    private MetacardMarshaller metacardMarshaller;
+
+    private SlotTypeHelper slotTypeHelper = new SlotTypeHelper();
+
+    private InternationalStringTypeHelper internationalStringTypeHelper =
+            new InternationalStringTypeHelper();
+
+    public void init() throws PrivilegedActionException {
+        Security.runAsAdminWithException(() -> {
+            if (!federationAdminService.getLocalRegistryIdentityMetacard()
+                    .isPresent()) {
+                createIdentityNode();
+            }
+            return null;
+        });
+
+    }
+
+    private void createIdentityNode() throws FederationAdminException {
+
+        String registryPackageId = RegistryConstants.GUID_PREFIX + UUID.randomUUID()
+                .toString()
+                .replaceAll("-", "");
+        RegistryPackageType registryPackage = RIM_FACTORY.createRegistryPackageType();
+        registryPackage.setId(registryPackageId);
+        registryPackage.setObjectType(RegistryConstants.REGISTRY_NODE_OBJECT_TYPE);
+
+        ExtrinsicObjectType extrinsicObject = RIM_FACTORY.createExtrinsicObjectType();
+        extrinsicObject.setObjectType(RegistryConstants.REGISTRY_NODE_OBJECT_TYPE);
+
+        String extrinsicObjectId = RegistryConstants.GUID_PREFIX + UUID.randomUUID()
+                .toString()
+                .replaceAll("-", "");
+        extrinsicObject.setId(extrinsicObjectId);
+
+        String siteName = SystemInfo.getSiteName();
```

Comment:
if for some reason the sitename, base url, or version is blank, do you need to handle that? Shouldn't happen since those are system properties.